internal/datacoord/meta.go

```diff
@@ -613,6 +614,19 @@ func (m *meta) GetSegments(segIDs []UniqueID, filterFunc SegmentInfoSelector) []
 	return result
 }
 
+func (m *meta) GetSegmentInfos(segIDs []UniqueID) []*SegmentInfo {
+	m.segMu.RLock()
+	defer m.segMu.RUnlock()
+	var result []*SegmentInfo
+	for _, id := range segIDs {
+		segment := m.segments.GetSegment(id)
+		if segment != nil {
+			result = append(result, segment)
+		}
+	}
+	return result
+}
+
```

Comment:
Please use SelectSegments instead, no need to add an extra method for this.
```golang
// Might look like this.
segmentSet := typeutil.UniqueSet(segmentIDs)
SelectSegments(SegmentFilterFunc(info *SegmentInfo) bool {
    segmentSet.Contains(info.ID)
})
```

Reply:
The SelectSegments function iterates through all segments to select the matching ones. For cases where the segmentID is determined, the GetSegment function is faster as it directly retrieves the key-value from the map.